.github/workflows/docs.yml

```diff
@@ -13,32 +13,40 @@
 
 name: Documentation
 on:
-  pull_request:
-    types: [opened, reopened, synchronize]
+  pull_request_target:
+    types: [opened, reopened, synchronize] # Handles forked PRs
   push:
   merge_group:
     types: [checks_requested]
+
 jobs:
   docs-build:
     name: Build documentation
     runs-on: ubuntu-latest
     permissions:
       pull-requests: write
     steps:
-      - name: Checkout repository
+      - name: Checkout repository (Handle all events)
         uses: actions/checkout@v4.2.2
+        with:
+          ref: ${{ github.head_ref || github.event.pull_request.head.ref || github.ref }}
```

Comment:
Can you add a comment here please? I'm guessing this handles the different event types like pull_request_target vs push.